### ISSUE TITLE
GitHub Actions: Ask for a preliminary update when issue is assigned

### ISSUE THREAD

**Issue by Aveline-art** [hackforla/website#2155]
### Overview

As a developer on this team, we need to update our issue once we have been assigned them. This way we will have made our first contribution and the team is aware on our ability to deliver on the issue. This action should be placed in the [Issue Trigger](https://github.com/hackforla/website/blob/gh-pages/.github/workflows/issue-trigger.yml) workflow and should run in parallel with the other jobs in the file.

### Psudo-code
- [x] Trigger: On issue assignment
  - [ ] If the issue contains a 'role: front end', 'role: back end', 'role: design', or 'role: user research' label:
    - [ ] Post a comment asking for an update. This comment should be similar to [update-instructions-template.md](https://github.com/hackforla/website/blob/gh-pages/github-actions/add-update-label-weekly/update-instructions-template.md), with a few key differences:
      - [x] Before the numeric sequence, the intro should read something like: "Thank you for taking on this issue! Before you begin, please answer the questions below:
      - [x] The numeric sequence should only contain availability and eta.
      - [x] Include a reminder that states that an second issue cannot be taken until this closes or is returned back to the backlog.
      - [x] Add a brief and nice closing statement such as, "Thank you!", or "You are awesome!"
      - [x] Make any other edits such that the comment would make sense.

### Checks

- [ ] When an issue is assigned and the issue contains no labels: does nothing.
- [ ] When an issue is assigned with both 'role: front end' and 'role: design': posts the comment.
- [ ] When an issue is assigned with 'role: back end': posts the comment.

### Resources/Instructions
[update-instructions-template.md](https://github.com/hackforla/website/blob/gh-pages/github-actions/add-update-label-weekly/update-instructions-template.md)

Never done GitHub Actions (GHA)? [Start here!](https://docs.github.com/en/actions)

Note that you might want to do something outside the scope of the above psudo-code. If so, be sure to leave comments in your PR or this issue that justifies your reasoning. If you feel you need guidance, be sure to reach out! We cannot foresee whether this issue is solvable, or what hard decisions have to be made, but we would love to hear and help you!

#### Additional resources:
[Events that trigger workflows](https://docs.github.com/en/actions/reference/events-that-trigger-workflows)
[Workflow syntax for GitHub Actions](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions)
[actions/github-script](https://github.com/actions/github-script)
[GitHub RESTAPI](https://docs.github.com/en/rest)
[GitHub GraphQL](https://docs.github.com/en/graphql)

**Comment by usaboo** [comment]
ETA - Thursday(March 3rd, 2022)
Availability - Tuesday (9 - 11PM EST)
Availability - Wednesday (9 - 11PM EST)

**Comment by usaboo** [comment]
Provide Update
1. Progress - 1st task completed.
2. Blockers - I do understand what I am supposed to do but I do not know, I have figured out the first task but that's about it.
3. Availability - Wednesday(4PM - 6PM EST), Thursday(4PM - 6PM EST), Friday(4PM - 6PM EST)
4. ETA - Sunday 13th, March 5PM EST 

### Taking much longer than expected, I have a few blockers in that I don't know if I know enough Javascript to tackle this issue. I will come to the doubts session to for help. Any help is appreciated. I will keep on working on this for this week, setting myself the deadline of Sunday's(13th March, 22) developer meeting.

**Comment by JessicaLucindaCheng** [comment]
@usaboo Please give a progress report in a comment in this issue. See what you need to include in your progress report and more information in [section 2.5 Reporting progress on your issue in CONTRIBUTING.md](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md#25-reporting-progress-on-your-issue)

**Comment by ExperimentsInHonesty** [comment]
- There was a comment from Saumil in the PR https://github.com/hackforla/website/pull/2962#pullrequestreview-989699924

> Additionally, I noticed that some of the functions such as getTimeline(), obtainLabels() and postComment() are being declared at multiple places in our codebase. It would be more efficient if we could reuse them, for instance by placing them in utils folder and then calling the function where required. We can make changes here (if you want) or have a new refactoring issue to implement this. Will let you and the merge team decide that since I won't be available going forward. Thanks for making it work.

Did this happen in the PR or was there a new issue made for it?

Please add the answer and link any relevant issues.

**Comment by jdingeman** [comment]
> @usaboo Hi Utkarsh, sorry for the delay in my review and thank you for making great progress on this issue. I understand you have been working on this for quite sometime and your contribution is really commendable. Just ran the tests in my repo and I couldn't find any gaps in the functionality to be implemented, so your PR has my approval.
> 
> Additionally, I noticed that some of the functions such as getTimeline(), obtainLabels() and postComment() are being declared at multiple places in our codebase. It would be more efficient if we could reuse them, for instance by placing them in utils folder and then calling the function where required. We can make changes here (if you want) or have a new refactoring issue to implement this. Will let you and the merge team decide that since I won't be available going forward. Thanks for making it work.

@ExperimentsInHonesty, I just checked the repo and it looks like these functions are still being declared in multiple places, so I think it'd be best that we write a new issue to consolidate these functions in a utils folder like Saumil said. It'll require some investigation across the .js files as there are some functions like obtainLabels() in check-label-preliminary-update.js that does not take a parameter, whereas the obtainLabels(context) in obtain-labels.js does take a parameter despite being essentially the same function. This is plenty of content for a new medium - large issue. See draft [here](https://github.com/hackforla/website/issues/3750)
